server/src/main/java/org/spine3/server/entity/AbstractEntity.java

```diff
@@ -181,6 +181,49 @@ private static IllegalStateException noSuchConstructor(String entityClass, Strin
         }
     }
 
+    /**
+     * Updates the state of the entity.
+     *
+     * <p>The new state must be {@linkplain #validate(Message) valid}.
+     *
+     * @param state the new state to set
+     * @throws IllegalStateException
+     *                if the passed state is not {@linkplain #validate(S) valid}
+     */
+    protected void updateState(S state) {
+        validate(state);
+        injectState(state);
+    }
+
+    /**
+     * Verifies whether the new state is valid.
+     *
+     * <p>Default implementation does nothing always returning {@code true}.
+     * Derived classes may provide custom validation logic overriding this method.
```

Comment:
Let's insert `by`, so that it will become `Derived classes may provide custom validation logic by overriding this method.`